File: 백준/Bronze/10708. 크리스마스 파티/크리스마스 파티.py
import sys
input = lambda: sys.stdin.readline().rstrip()


def main():
    N = int(input())
    M = int(input())
    friend = [0]*N
    target = list(map(int, input().split()))
    for m in range(M):
        tmp, miss = 0, 0
        lst = list(map(int, input().split()))
        for n in range(N):
            if lst[n] == target[m]:
                friend[n] += 1
            else:
                miss += 1
        friend[target[m]-1] += miss
    print(*friend, sep = '\n')


if __name__ == '__main__':
    main()

File: 백준/Bronze/10708. 크리스마스 파티/크리스마스 파티.c
#include <stdio.h>

int main() {
    int N, M;
    scanf("%d %d", &N, &M);

    int friend[N];
    for (int i = 0; i < N; i++) {
        friend[i] = 0;
    }
    int target[M];
    for (int i = 0; i < M; i++) {
        scanf("%d", &target[i]);
    }

    for (int m = 0; m < M; m++) {
        int miss = 0;
        int lst[N];
        for (int n = 0; n < N; n++) {
            scanf("%d", &lst[n]);
            if (lst[n] == target[m]) {
                friend[n]++;
            } else {
                miss++;
            }
        }
        friend[target[m] - 1] += miss;
    }
    for (int i = 0; i < N; i++) {
        printf("%d\n", friend[i]);
    }
    return 0;
}